docs/project.md

```diff
@@ -2,4 +2,154 @@
 
 A project is a an app that contains React code and has a dependency on `react-native`.
 
-Projects can provide additional properties to alter the CLI behavior (e.g. a project is located in a non-standard place) as well as change the configuration for each of the dependencies.
+Projects can provide additional properties to alter the CLI behavior, such as custom location of React Native files (this is useful when running RNTester from source) or a non-standard location of a project (useful when working in a brownfield app).
+
+## How does it work?
+
+A project can define a `react-native.config.js` at the root with custom configuration to be picked up by the CLI.
+
+For example, below configuration informs CLI of the additional assets to link and about a custom project location.
+
+```js
+module.exports = {
+  project: {
+    ios: {
+      project: './CustomProject.xcodeproj'
+    }
+  },
+  assets: ['./assets']
+};
+```
+
+You can check all available options below.
+
+## Project interface
+
+```ts
+type UserConfigT = {
+  reactNativePath: ?string,
+  project: {
+    android?: ProjectParamsAndroidT,
+    ios?: ProjectParamsIOST,
+    [key: string]: any,
+  },
+  assets: string[],
+  platforms: PlatformT,
+  dependencies: {
+    [key: string]: {
+      name: string,
+      root: string,
+      platforms: {
+        [key: string]: PlatformSettingsT
+      },
+      assets: string[],
+      hooks: {
+        [key: string]: string
+      }
+    },
+  },
+  commands: CommandT[]
+};
+```
+
+### reactNativePath
+
+A custom path to React Native, in case `require('react-native')` would throw. Useful when running
+React Native from a (custom) source.
```

Comment:
I wonder do we still need this option? I can imagine `yarn link` may fail but otherwise?

Reply:
Well, we need it to run RNTester because there's no `react-native` package - our default resolver will not work. I wanted to provide a gentle solution to this instead of providing half-baked support.